### ISSUE TITLE
service/ecr: waiter comparator type mismatch from ImageScanCompleteWaiter 

### ISSUE THREAD

**Issue by saj** [aws/aws-sdk-go-v2#1063]
Confirm by changing [ ] to [x] below to ensure that it's a bug:
- [x] I've gone though the [API reference](https://docs.aws.amazon.com/sdk-for-go/v2/api/)
- [x] I've checked [AWS Forums](https://forums.aws.amazon.com) and [StackOverflow](https://stackoverflow.com/questions/tagged/aws-sdk-go) for answers
- [x] I've searched for [previous similar issues](https://github.com/aws/aws-sdk-go-v2/issues) and didn't find any solution
  
**Describe the bug**

[`ecr.ImageScanCompleteWaiter`](https://pkg.go.dev/github.com/aws/aws-sdk-go-v2/service/ecr#ImageScanCompleteWaiter) appears to be unusable; it fails with the following error:

```
waiter comparator expected string value got types.ScanStatus
```

I believe this is a bug because the [value under comparison](https://github.com/aws/aws-sdk-go-v2/blob/service/ecr/v1.0.0/service/ecr/api_op_DescribeImageScanFindings.go#L387) will never be of string type.

**Version of AWS SDK for Go?**

1.0.0

**Version of Go (`go version`)?**

go version go1.15.6 darwin/amd64

**To Reproduce (observed behavior)**

It is difficult to share a complete working example; here is the general gist of what I am trying to do:

```go
import (
	"context"

	"github.com/aws/aws-sdk-go-v2/service/ecr"
	ecrtypes "github.com/aws/aws-sdk-go-v2/service/ecr/types"
)

c := ecr.NewFromConfig(...)
p := ecr.DescribeImageScanFindingsInput{
	RepositoryName: aws.String("some-repo-name"),
	ImageId:        &ecrtypes.ImageIdentifier{ImageTag: aws.String("some-unique-image-tag")},
}
w := ecr.NewImageScanCompleteWaiter(c)
if err := w.Wait(context.TODO(), &p, 10*time.Minute); err != nil {
	// handle error
}
```

**Expected behavior**

The [`Wait()`](https://pkg.go.dev/github.com/aws/aws-sdk-go-v2/service/ecr#ImageScanCompleteWaiter.Wait) method should eventually return no error.

**Additional context**

Separately, it seems that in order to use the provided paginator and waiter for this type, it is necessary to make the same request over the network twice:  once in order to unblock the waiter, then again in order to begin iterating through the paginated responses.  Is this a known limitation, or have I missed something?

**Comment by skotambkar** [comment]
Thanks for reporting the bug. We will investigate the issue and create a fix. 

Could you also clarify what your use case is when using waiter along with paginator?  If you have an example of how you would be doing waiters with paginators currently, that would make it easier for us to understand.

**Comment by saj** [comment]
Sure.  We provide our staff with a small command-line utility that can be used to check newly built container images for known vulnerabilities in packaged software.  This tool works by uploading the image to AWS ECR and blocking for scan results.  The results are printed when they eventually become available.  (We have one repository set aside for this purpose; the repository is configured with `ScanOnPush` and an aggressive lifecycle policy that will remove images after about one day.  The copy on ECR is effectively ephemeral.)  We were already using AWS ECR for long-term production image storage (and vulnerability reporting), so this little command-line utility felt like a natural extension.

After uploading an image, it can take a moment (maybe a minute or two, though I have not measured) before the scan results become available from ECR.  We block here.  Previously, this block was implemented using our own interruptible sleep-and-retry implementation.  Upon seeing the GA release of aws-sdk-go-v2, I decided to port this trivial program over and familiarise myself with your new API.  This was where I attempted to make use of the `ecr.ImageScanCompleteWaiter`.

Once we know the scan is complete, we then begin iterating through the scan results.  Depending on the nature of the image, there may be many findings in the scan -- too many to fit within a single page of results.  Usually, we are interested in the most severe findings (say, those of `FindingSeverityHigh` or more).  IIRC, the [`DescribeImageScanFindings`](https://docs.aws.amazon.com/AmazonECR/latest/APIReference/API_DescribeImageScanFindings.html) action is not documented to return findings in any specific order, so we must be careful to iterate through all pages and filter/sort locally to be assured that the most severe findings are surfaced to the user.  This was where I had intended to make use of the `DescribeImageScanFindingsPaginator`.

Previously, on v1, our sleep-and-retry implementation handled both waits and page iteration.  The response we received from the `DescribeImageScanFindings` action would first be tested for `status == COMPLETE`; if that were true, would we then **reuse** that response to kick off the page walk.  There was no need for a redundant request.  (This approach is still viable on v2, of course, but I was hoping to make use of the provided waiter/paginator now that they exist.)

Does that help?
